diff --git a/include/mscclpp/poll_device.hpp b/include/mscclpp/poll_device.hpp
index 9ad116f..33325c8 100644
--- a/include/mscclpp/poll_device.hpp
+++ b/include/mscclpp/poll_device.hpp
@@ -14,10 +14,10 @@
 #define __assert_fail(__assertion, __file, __line, __function) ;
 #else  // !defined(NDEBUG)
 #if defined(MSCCLPP_DEVICE_HIP)
-extern "C" __device__ void __assert_fail(const char *__assertion, const char *__file, unsigned int __line,
+extern "C" __host__ __device__ void __assert_fail(const char *__assertion, const char *__file, unsigned int __line,
                                          const char *__function);
 #else   // !defined(MSCCLPP_DEVICE_HIP)
-extern "C" __device__ void __assert_fail(const char *__assertion, const char *__file, unsigned int __line,
+extern "C" __host__ __device__ void __assert_fail(const char *__assertion, const char *__file, unsigned int __line,
                                          const char *__function) __THROW;
 #endif  // !defined(MSCCLPP_DEVICE_HIP)
 #endif  // NDEBUG
