From 550e5c626f9f86ac8100155d767f52fc8c4dc815 Mon Sep 17 00:00:00 2001
From: Alkaid <zgf574564920@gmail.com>
Date: Tue, 26 Mar 2024 14:50:39 -0700
Subject: [PATCH] Expose internal GEMM fields for pllm

---
 include/cutlass/gemm/device/gemm.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/cutlass/gemm/device/gemm.h b/include/cutlass/gemm/device/gemm.h
index f0226354..805e3d2c 100644
--- a/include/cutlass/gemm/device/gemm.h
+++ b/include/cutlass/gemm/device/gemm.h
@@ -346,8 +346,6 @@ class Gemm {
     }
   };
 
-private:
-
   /// Kernel parameters object
   typename GemmKernel::Params params_;
 
-- 
2.44.0

